File: topological_sorting/go/topological_sorting_test.go
package topological_sorting

import (
	"reflect"
	"testing"
)

var tests = []struct {
	graph Graph
	sorts [][]int
}{
	{
		Graph{
			7,
			true,
			[]Edge{
				{0, 1},
				{1, 3},
				{1, 2},
				{2, 4},
				{0, 2},
				{4, 3},
				{6, 0},
				{6, 5},
				{2, 5},
				{5, 4},
			},
		},
		[][]int{
			[]int{6, 0, 1, 2, 5, 4, 3},
		},
	},
}

func Test(t *testing.T) {
	for i, test := range tests {
		if s := Sort(test.graph); !contains(test.sorts, s) {
			t.Errorf("In test #%d with input graph %+v, expected topological sort to be one of %v, was %v",
				i, test.graph, test.sorts, s)
		}
	}
}

func contains(sorts [][]int, sort []int) bool {
	for _, s := range sorts {
		if reflect.DeepEqual(s, sort) {
			return true
		}
	}
	return false
}

File: topological_sorting/typescript/topological_sorting_test.ts
import * as assert from 'assert';
import {Graph} from './adjacency_list';
import sort from './topological_sorting';

interface Test {
    graph: Graph;
    sorts: number[][];
}

const tests: Test[] = [
    {
        graph: {
            vertexCount: 7,
            directed: true,
            edges: [
                {x: 0, y: 1},
                {x: 1, y: 3},
                {x: 1, y: 2},
                {x: 2, y: 4},
                {x: 0, y: 2},
                {x: 4, y: 3},
                {x: 6, y: 0},
                {x: 6, y: 5},
                {x: 2, y: 5},
                {x: 5, y: 4},
            ],
        },
        sorts: [
            [6, 0, 1, 2, 5, 4, 3],
        ],
    },
];

const deepEqual = (a: any, b: any) => {
    try {
        assert.deepStrictEqual(a, b);
        return true;
    } catch (e) {
        return false;
    }
};

const contains = (list: number[][], item: number[]) =>
    list.some(listItem => deepEqual(listItem, item));

function runTest(test: Test): void {
    assert.ok(contains(test.sorts, sort(test.graph)));
}

tests.forEach(runTest);
